pkg/karmadactl/cmdinit/kubernetes/node.go

```diff
@@ -31,6 +31,12 @@ func (i *CommandInitOption) getKarmadaAPIServerIP() error {
 	if err != nil {
 		return err
 	}
+	if len(masterNodes.Items) == 0 {
```

Comment:
Perhaps we can add a comment to account for this piece of code.

I just tested it in v1.24.6 kubernetes that work fine.